src/pages/index.js

```diff
@@ -80,64 +74,185 @@ const addCardPopup = new PopupWithForm(
-// 	}
-// }
-
-function renderCard(cardData) {
-	const card = new Card(cardData, '#card-template', handleImageClick);
-	const cardElement = card.getView();
-	cardSection.addItem(cardElement);
+function handleImageClick(name, link) {
+	popupImageModal.open({ name, link });
 }
 
-function handleOverlayClose(e) {
-	if (e.target.classList.contains('modal_opened')) {
-		popupImageModal.close();
-	}
+//  
+// Form Validation
+//  
+
+const addEditValidator = new FormValidator(config, editFormElement);
+addEditValidator.enableValidation();
+
+const addCardValidator = new FormValidator(config, addFormElement);
+addCardValidator.enableValidation();
+
+const updateAvatarValidator = new FormValidator(config, avatarFormElement);
+updateAvatarValidator.enableValidation();
+
+//  
+// User Info
+//  
+
+const userInfo = new UserInfo({
+	nameSelector: '.profile__title',
+	jobSelector: '.profile__description',
+	userAvatar: '.profile__image',
+});
+
+// 
+// Functions
+//  
+
+function createCard(cardData) {
+	const card = new Card(
+		cardData,
+		'#card-template',
+		handleImageClick,
+		handleTrashClick,
+		handleLikeClick,
+	);
+	cardSection.addItem(card.generateCard());
 }
 
-function handleImageClick(name, link) {
-	popupImageModal.open(name, link);
+// ADD CARD AND EDIT PROFILE
+
+function handleAddCardFormSubmit(data) {
+	addCardPopup.submitButtonState(true, "Saving...");
+	api
+		.addCard(data)
+		.then((data) => {
+			createCard(data);
+		})
+		.then(() => {
+			addCardPopup.close();
+		})
+		.catch((err) => {
+			console.log(err);
+		})
+		.finally(() => {
+			addCardPopup.submitButtonState(false);
+		});
 }
 
-function handleProfileEditSubmit(value) {
-	userInfo.setUserInfo(value);
-	editProfilePopup.close();
+function handleProfileEditSubmit(data) {
+	editProfilePopup.submitButtonState(true, "Saving...");
+	api
+	.updateEditProfile(data)
+		.then((data) => {
+			userInfo.setUserInfo(data);
+			editProfilePopup.close();
+		})
+		.catch((err) => {
+			console.log(err);
+		})
+		.finally(() => {
+			editProfilePopup.submitButtonState(false);
+		});
+}
+
+//  
+//  Avatar
+//  
+
+const avatarPopup = new PopupWithForm(
+	'#avatar-modal',
+	handleAvatarFormSubmit
+);
+avatarPopup.setEventListeners();
+
+function handleAvatarFormSubmit(data) {
+	avatarPopup.submitButtonState(true, "Saving...");
+	api
+	.updateAvatar(data)
+	  .then((data) => {
+		// userInfo.setUserInfo(data);
+		userInfo.setAvatar(data.avatar);
+		avatarPopup.close();
+	  })
+	  .catch((err) => {
+		console.log(err);
+	  })
+	  .finally(() => {
+		avatarPopup.submitButtonState(false);
+	  });
+}
+
+const avatarButton = document.querySelector(".avatar__edit_button");
+avatarButton.addEventListener("click", () => {
+  updateAvatarValidator.toggleButtonState();
+  avatarPopup.open();
+});
+
+//  
+//  Like Card
+//
+
+function handleLikeClick(card) {
+  if (card.isLiked) {
+	api
+	  .unlikeCard(card.id)
+	  .then(() => {
+		card.updateIsLiked(false)
+	  });
```

Comment:
You need to `catch` possible errors at the very end of any server request.

**Please, correct it everywhere**